src/init.cpp

```diff
@@ -1108,6 +1111,44 @@ bool AppInitLockDataDirectory()
     return true;
 }
 
+/**
+ * Once initial block sync is finished and no change in block height or queued downloads,
+ * sync utxo state to protect against data loss
+ */
+static void SyncCoinsTipAfterChainSync(const NodeContext& node)
+{
+    LOCK(node.chainman->GetMutex());
+    if (node.chainman->IsInitialBlockDownload()) {
+        LogDebug(BCLog::COINDB, "Node is still in IBD, rescheduling post-IBD chainstate disk sync...\n");
+        node.scheduler->scheduleFromNow([&node] {
+            SyncCoinsTipAfterChainSync(node);
+        }, SYNC_CHECK_INTERVAL);
+        return;
+    }
```

Comment:
Point taken for moving the explicit lock after this check, since the lock is taken in `IsInitialBlockDownload()`.

However, this check only runs once every 30 seconds. I don't see how it could possibly affect responsiveness of the software.  It is a very fast check I would assume on the order of microseconds every 30 seconds.